runelite-client/src/main/java/net/runelite/client/plugins/opponentinfo/OpponentInfoPlugin.java

```diff
@@ -134,33 +137,46 @@ else if (worldType.contains(WorldType.LEAGUE))
 	@Subscribe
 	public void onInteractingChanged(InteractingChanged event)
 	{
-		if (event.getSource() != client.getLocalPlayer())
-		{
-			return;
-		}
+		Actor player = client.getLocalPlayer(), source = event.getSource(), target = event.getTarget();
 
-		Actor opponent = event.getTarget();
-
-		if (opponent == null)
+		if (source == player)
 		{
-			lastTime = Instant.now();
-			return;
+			if (target != null)
+			{
+				lastOpponent = target;
+				lastTime = null;
+			}
+			else if (lastOpponent != null
+				&& lastOpponent.getInteracting() != player)
+			{
+				lastTime = Instant.now();
```

Comment:
It looks like this line is not covered in the added tests. What case does this cover?

Reply:
This prevents the hp bar disappearing in the case where the player attacks an npc, then stops attacking anything, but the npc hasn't stopped attacking the player.  There actually is a test for it on line 184.

Reply:
That isn't what this condition is written to be though, nor is it what that test is testing.

This condition, combined with all the parent conditions, works out to be:

```java
if (source == player && target == null && lastOpponent != null && lastOpponent.getInteracting() != player)
```

That is, when the player stops interacting with an NPC which _is not_ attacking the player. That is why, despite the test being written to test the scenario you described, this line is not being covered by tests. I highly recommend running the tests in IntelliJ using the "Run Test with Coverage" button and checking to see which lines of the plugin (most importantly the changed lines) have test coverage. That is how I identified that this `lastTime = Instant.now();` call is never executed by the tests.

Reply:
Sorry, I explained poorly.  What I meant was exactly what you described, I was just explaining what happens when the condition is false.  It is designed so that in the case I mentioned, where the player stops attacking an npc, but the npc is still attacking the player, the condition is false, so the hp bar stays visible.  That's why when—as you've described—the player stops interacting with an npc that isn't attacking the player, the condition is true.  The result of that is, as you would expect, the hp bar is set to disappear after WAIT time.  In the test, I was testing that case where the condition is false.  I've added a test where the condition is true now.